components/brave_rewards/resources/extension/brave_rewards/background/reducers/rewards_panel_reducer.ts

```diff
@@ -25,6 +25,68 @@ const getWindowId = (id: number) => {
   return `id_${id}`
 }
 
+function pollTwitchPage (tab: chrome.tabs.Tab, tabId: number, publisherBlob:string): void {
+  let url = tab.url
+  // use an interval here to monitor when the DOM has finished
+  // generating. clear after the data is present.
+  // Check every second no more than 'limit' times
+  // clear the interval if panel closes
+  const markupMatch = '<figure class=\"tw-avatar tw-avatar--size-36\">' +
+                      '<div class=\"tw-border-radius-medium tw-overflow-hidden\">' +
+                      '<img class=\"tw-avatar__img tw-image\" alt=\"'
+  let obtained = false
+  // run once to see if we already have it
```

Comment:
as mentioned below, i don't think this will work because the callback is async whereas https://github.com/brave/brave-core/pull/888#diff-fc21efee53a5e1ba98eb186ef1e788a8R55 is sync